react/src/views/Transfers/CreateTransferAgreement/components/CreateTransferAgreement.tsx

```diff
@@ -0,0 +1,269 @@
+
+export interface ICreateTranferAgreementProps {
+  currentOrganisation: IBasesForOrganisationData;
+  partnerOrganisationsWithTheirBasesData: IBasesForOrganisationData[];
+  onSubmitCreateTransferAgreementForm: (
+    transferAgreementFormValues: ITransferAgreementFormData,
+  ) => void;
+}
+
+function CreateTransferAgreement({
+  currentOrganisation,
+  partnerOrganisationsWithTheirBasesData,
+  onSubmitCreateTransferAgreementForm,
+}: ICreateTranferAgreementProps) {
+  const sourceBasesForDropdownGroups: Array<IDropdownOption> | undefined =
+    currentOrganisation?.bases.map((base) => ({
+      label: base.name,
+      value: base.id,
+    }));
+
+  const partnerOrganisationsForDropdownGroups: Array<IDropdownOption> | undefined =
+    partnerOrganisationsWithTheirBasesData?.map((organisation) => ({
+      label: organisation.name,
+      value: organisation.id,
+    }));
+
+  const onSubmit: SubmitHandler<ITransferAgreementFormData> = (data) => {
+    // eslint-disable-next-line no-console
+    onSubmitCreateTransferAgreementForm(data);
+  };
+
+  const {
+    handleSubmit,
+    control,
+    register,
+    resetField,
+    watch,
+    setValue,
+    formState: { errors, isSubmitting },
+  } = useForm<ITransferAgreementFormData>({
+    resolver: zodResolver(TransferAgreementFormDataSchema),
+  });
+
+  const [basesOptionsForPartnerOrg, setBasesOptionsForPartnerOrg] = useState<IDropdownOption[]>([]);
+  const [validUntilMinDate, setValidUntilMinDate] = useState("");
+  const partnerOrganisation = watch("partnerOrganisation");
+  const validFrom = watch("validFrom");
+
+  useEffect(() => {
+    if (partnerOrganisation != null) {
+      const partnerBasesDataForCurrentOrganisation = partnerOrganisationsWithTheirBasesData.find(
+        (organisation) => organisation.id === partnerOrganisation.value,
+      );
+
+      setBasesOptionsForPartnerOrg(
+        () =>
+          partnerBasesDataForCurrentOrganisation?.bases?.map((base) => ({
+            label: base.name,
+            value: base.id,
+          })) || [],
+      );
+
+      resetField("partnerOrganisationSelectedBases");
+      // Put a default value for partnerOrganisationSelectedBases when there's only one option
+      if (partnerBasesDataForCurrentOrganisation?.bases.length === 1) {
+        setValue("partnerOrganisationSelectedBases", [
+          {
+            label: partnerBasesDataForCurrentOrganisation?.bases[0].name,
+            value: partnerBasesDataForCurrentOrganisation?.bases[0].id,
+          },
+        ]);
+      }
+    }
+    if (currentOrganisation?.bases.length === 1) {
+      setValue("currentOrganisationSelectedBases", [
+        {
+          label: currentOrganisation?.bases[0].name,
+          value: currentOrganisation?.bases[0].id,
+        },
+      ]);
+    }
+
+    if (validFrom) {
+      setValidUntilMinDate(
+        new Date(new Date(validFrom).setDate(new Date(validFrom).getDate() + 1))
```

Comment:
Why are there three `new Date` calls? :sweat_smile:

Reply:
@pylipp 😅 Let me refactor this with the `addDays` function of `date-fns`